### ISSUE TITLE
Print Plate in BS Overwrites Files in Internal Storage

### ISSUE THREAD

**Issue by CCS86** [bambulab/BambuStudio#1107]
Even when the new job has a unique name, initiating a print from BS overwrites the last user job in the internal memory.

Why are sample files preserved and actual user jobs overwritten? Why can we only have one user job stored on the printer for fast access?

**Comment by lanewei120** [comment]
Hello

thanks for reporting this bug

which printer are you using?X1/X1C or P1P
and which studio version and printer firmware are you using?

and did you have a sdcard plugged in when printing?

**Comment by CCS86** [comment]
It is an X1C

Yes, I have an SD card installed for time lapses.

**Comment by lanewei120** [comment]
the internal memory only saves the current print job
and the history jobs will be saved in sdcard
we can find them on printer and print again from printer

is this what you want?

**Comment by CCS86** [comment]
Is there a reason that history jobs cannot be stored on the internal memory?

Storing on the SD card would be okay, but that is not happening. 

Maybe the fact that my primary computer is on our LAN and the X1C is on our WIFI network, and these are on different subnets, is causing files not to be stored correctly on the SD card? I'm still not sure I understand that network limitation.

When you send a print job the printer itself should move the last print job to the SD card.

**Comment by lanewei120** [comment]
@sodachen22 Could you give more explanations on this logic on printer side?

**Comment by sodachen22** [comment]
@CCS86, In the latest firmware, the 3MF for the printed job is in SD card. And there is only one history printed job through cloud is kept in internal memory.

**Comment by CCS86** [comment]
> @CCS86, In the latest firmware, the 3MF for the printed job is in SD card. And there is only one history printed job through cloud is kept in internal memory.


It is not though. 

The only print file on my SD card is a gcode file I manually copied there via computer. I have sent many jobs via BS and none appear on the SD card.

**Comment by sodachen22** [comment]
@CCS86 Are your firmware are latest and the SD card have enough free space?

**Comment by CCS86** [comment]
> @CCS86 Are your firmware are latest and the SD card have enough free space?

Yes.

**Comment by sodachen22** [comment]
The lastest firmwre is still in beta testing.

**Comment by CCS86** [comment]
Before the 1.04 FW release, my printer started storing **some** print jobs to the SD card. But for some reason, most of them are not being retained.

Here is my SD card contents:

![image](https://user-images.githubusercontent.com/12439543/217881909-552b40e8-0c0b-4d28-8c8c-098e59ecefc8.png)

But my print job history has over 40 prints in it.

Even since updating to 1.04, the behavior has not changed and my most recent job has not been moved to the SD card. It is still sitting in the internal storage, with the wrong name:

![image](https://user-images.githubusercontent.com/12439543/217883015-5c97b267-8000-433e-9e29-064220da217b.png)

**Comment by mizone0426** [comment]
Is there any history print in "Print Cache" sheet?

**Comment by CCS86** [comment]
> Is there any history print in "Print Cache" sheet?

Not for me. It is always empty.

I think the only time jobs go there is if it is already printing and you send a new job.

**Comment by sodachen22** [comment]
@CCS86  There is an option for print cache here, is it enabled? Is there cache directory created in SD card?
![image](https://user-images.githubusercontent.com/112450380/218236064-ebab459b-bdb0-4d7c-8479-ebc9b896c639.png)

**Comment by sodachen22** [comment]
> I think the only time jobs go there is if it is already printing and you send a new job.
Yes, this function is caching 3MF of the printed job after "Cache cloud print file to MicroSD card" is enabled. It is not syncing print history of from the cloud service.

**Comment by CCS86** [comment]
> > Is there any history print in "Print Cache" sheet?
> 
> Not for me. It is always empty.
> 
> I think the only time jobs go there is if it is already printing and you send a new job.


I'm not near the printer right now, but in Bambu Handy and Bambu Studio, I do not have that option. Both end with "auto-recovery from step loss"

**Comment by sodachen22** [comment]
> I'm not near the printer right now, but in Bambu Handy and Bambu Studio, I do not have that option. Both end with "auto-recovery from step loss"
The option is only on the screen of the printer now. One of the reasons is Handy/Studio does not support browsing and printing 3MF on the SD card of the printer remotely.

**Comment by CCS86** [comment]
Even with that option enabled, my print jobs are not reliably accumulating on the SD card.

**Comment by sodachen22** [comment]
The total cache file number and size are limited. Would you please take a look at the description of this feature in the release notes?
https://wiki.bambulab.com/en/x1/manual/X1-X1C-AMS-firmware-release-history
![image](https://user-images.githubusercontent.com/112450380/222662410-73ce6665-10a3-4a36-beb4-d647467d0073.png)

**Comment by CCS86** [comment]
That's fine if you want to limit the "print cache" to 8 files, but why can we not have an option to keep a running history of **all** prints on the SD card, if we have plenty of space to do so?
